docs/howto/advanced/retry.md

```diff
@@ -64,22 +64,49 @@ between retries:
 ## Implementing your own strategy
 
 - If you want to go for a fully fledged custom retry strategy, you can implement your
-  own retry strategy (though we recommend always keeping a max_retry):
+  own retry strategy by returning a `RetryDecision` object from the
+  `get_retry_decision` method. This also allows to (optionally) change the priority,
+  the queue or the lock of the job. The time to wait between retries can be specified
+  with `retry_in` or alternatively with `retry_at`. If `None` is returned
+  from `get_retry_decision` then the job will not be retried.
+
+  ```python
+  import random
+  from procrastinate import Job, RetryDecision
 
+  class RandomRetryStrategy(procrastinate.BaseRetryStrategy):
+      max_attempts = 3
+      min = 1
+      max = 10
+
+      def get_retry_decision(self, *, exception:Exception, job:Job) -> RetryDecision:
+          if job.attempts >= max_attempts:
+              return RetryDecision(should_retry=False)
+
+          wait = random.uniform(self.min, self.max)
+
+          return RetryDecision(
+              retry_in={"seconds": 10}, # or retry_at (a datetime object)
+              priority=job.priority + 1, # optional
+              queue="another_queue", # optional
+              lock="another_lock", # optional
+          )
   ```
+
+- There is also a legacy `get_schedule_in` method that is deprecated an will be
```

Comment:
```suggestion
- There is also a deprecated `get_schedule_in` method, which will be
```

Reply:
Actually... Let's mention `get_schedule_in` but not the example. I don't think people need example of how to implement deprecated methods.

Reply:
Makes sense.